http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GKEInterpreter.scala

```diff
@@ -0,0 +1,390 @@
+          config.monitorConfig.nodepoolCreate.interval,
+          config.monitorConfig.nodepoolCreate.maxAttempts
+        )
+        .compile
+        .lastOrError
+      _ <- if (lastOp.isDone) logger.info(s"Nodepool creation operation has finished for nodepool ${nodepoolId}")
+      else
+        logger.error(s"Create nodepool operation has failed or timed out for nodepool $nodepoolId") >> F
+          .raiseError[Unit](NodepoolCreationException(nodepoolId))
+      _ <- nodepoolQuery.updateStatus(nodepoolId, NodepoolStatus.Running).transaction
+    } yield ()
+
+    createNodepool.onError {
+      case e =>
+        for {
+          now <- nowInstant
+          _ <- logger.error(e)(s"Failed to create nodepool $nodepoolId")
+          _ <- nodepoolQuery.updateStatus(nodepoolId, NodepoolStatus.Error).transaction
+          _ <- appErrorQuery
+            .save(appId, KubernetesError(e.getMessage(), now, ErrorAction.CreateGalaxyApp, ErrorSource.Nodepool, None))
+            .transaction
+        } yield ()
+    }
+  }
+
+  def deleteNodepool(getAppResult: GetAppResult)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val nodepoolId = getAppResult.nodepool.id
+    val deleteNodepool = for {
+      ctx <- ev.ask
+      op <- gkeService.deleteNodepool(
+        NodepoolId(getAppResult.cluster.getGkeClusterId, getAppResult.nodepool.nodepoolName)
+      )
+      //TODO: refactor this to handle interupts
+      lastOp <- gkeService
+        .pollOperation(
+          KubernetesOperationId(getAppResult.cluster.googleProject, getAppResult.cluster.location, op),
+          config.monitorConfig.nodepoolDelete.interval,
+          config.monitorConfig.nodepoolDelete.maxAttempts
+        )
+        .compile
+        .lastOrError
+      _ <- if (lastOp.isDone) F.unit
+      else
+        logger.error(s"Delete nodepool operation has failed or timed out for nodepool ${nodepoolId}") >> F
+          .raiseError[Unit](NodepoolDeletionException(nodepoolId))
+      _ <- nodepoolQuery.markAsDeleted(nodepoolId, ctx.now).transaction
+    } yield ()
+
+    deleteNodepool.onError {
+      case e =>
+        for {
+          now <- nowInstant
+          _ <- logger.error(e)(s"Failed to delete nodepool $nodepoolId")
+          _ <- nodepoolQuery.updateStatus(nodepoolId, NodepoolStatus.Error).transaction
+          _ <- appErrorQuery
+            .save(getAppResult.app.id,
+                  KubernetesError(e.getMessage(), now, ErrorAction.DeleteGalaxyApp, ErrorSource.Nodepool, None))
+            .transaction
+        } yield ()
+    }
+  }
+
+  def deleteApp(getAppResult: GetAppResult)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    //TODO: this should actually monitor the deletion went through successfully before updating the status
+    val deleteApp = for {
+      now <- nowInstant
+      _ <- kubeService.deleteNamespace(getAppResult.cluster.getGkeClusterId,
```

Comment:
Need to make sure deleting the namespace actually destroys all resources created by the app

Reply:
I've verified that it deletes all things currently created at least. sounds like a good automation test too cc @xuscrbw